pkg/workloads/cortex/lib/client/tensorflow.py

```diff
@@ -258,19 +275,35 @@ def create_prediction_request(signature_def, signature_key, model_name, model_in
     prediction_request.model_spec.signature_name = signature_key
 
     for column_name, value in model_input.items():
-        shape = []
-        for dim in signature_def[signature_key]["inputs"][column_name]["tensorShape"]["dim"]:
-            shape.append(int(dim["size"]))
+        if signature_def[signature_key]["inputs"][column_name]["tensorShape"] == {}:
+            shape = "scalar"
+        elif signature_def[signature_key]["inputs"][column_name]["tensorShape"].get(
+            "unknownRank", False
+        ):
```

Comment:
nit: is this equivalent to `elif "unknownRank" in signature_def[signature_key]["inputs"][column_name]["tensorShape"]:`, and if so, do you prefer it?

Reply:
Yes, that is the equivalent. There's one little thing I noticed: `unknownRank` always seems to be set to `True` when no rank is specified, and so, I think it's also good to check the key's value. I'm not sure if there's a case when that's set to `False`.

Reply:
I could add a comment to describe the reason.

Reply:
That makes sense, yes, it is better to be safe and check the value too. Thanks!